hub/core/storage/provider.py

```diff
@@ -125,3 +127,13 @@ def flush(self):
         """Only needs to be implemented for caches. Flushes the data to the next storage provider.
         Should be a no op for Base Storage Providers like local, s3, azure, gcs, etc.
         """
+
+    def enable_autoflush(self):
+        self.autoflush = True
+
+    def disable_autoflush(self):
+        self.autoflush = False
+
+    def maybe_flush(self):
```

Comment:
`maybe_flush` appears quite confusing. Can we rename it to something else?

Reply:
What do you prefer? I can't think of anything better.

Reply:
naming is ok, but can you add a docstring to clarify?

Reply:
Yeah, this is anyway not user-facing. We can leave the name as is but a docstring is definitely needed.